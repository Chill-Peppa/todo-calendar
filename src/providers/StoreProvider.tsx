import React, { FC, useState } from 'react';
import { StoreContext } from '../hooks/useStore';

interface IStoreProvider {
  children: React.ReactNode;
}
export const StoreProvider: FC<IStoreProvider> = ({ children }) => {
  const [text, setText] = useState('');

  return (
    <StoreContext.Provider value={{ text, setText }}>
      {children}
    </StoreContext.Provider>
  );
};
