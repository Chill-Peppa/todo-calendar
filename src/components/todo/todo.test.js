import renderer from 'react-test-renderer';
import { StoreProvider } from '../../providers/StoreProvider';

import Todo from './todo';

it('should correctly render Todo', () => {
  const task = {
    id: 123,
    todo: 'test todo',
    isDone: false,
    date: '2022-01-01',
  };
  const i = 0;

  const tree = renderer
    .create(
      <StoreProvider>
        <Todo task={task} i={i} />
      </StoreProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
