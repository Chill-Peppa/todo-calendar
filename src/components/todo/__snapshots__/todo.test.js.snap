// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Todo 1`] = `
<li
  className="modal__deal"
>
  1. test todo
  <div
    className="modal__checkbox-container"
  >
    <input
      className="modal__checkbox"
      id="checkboxid-123"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="modal__label"
      htmlFor="checkboxid-123"
    />
    <img
      alt="delete"
      className="modal__delete-deal"
      onClick={[Function]}
      src="delete.svg"
    />
  </div>
</li>
`;
